iothub/device/src/Transport/Mqtt/OrderedTwoPhaseWorkQueue.cs

```diff
@@ -43,19 +45,29 @@ public Task CompleteWorkAsync(IChannelHandlerContext context, TWorkId workId)
         {
             if (!_incompleteQueue.Any())
             {
-                throw new IotHubException("Nothing to complete.", isTransient: false);
+                if (Logging.IsEnabled)
+                {
+                    Logging.Error(context, $"{nameof(CompleteWorkAsync)} called but there are no items in the queue to complete.", nameof(CompleteWorkAsync));
+                }
+
+                Debug.Fail($"{nameof(CompleteWorkAsync)} called but there are no items in the queue to complete.");
+                return TaskHelpers.CompletedTask;
             }
 
             IncompleteWorkItem incompleteWorkItem = _incompleteQueue.Peek();
```

Comment:
Can we guarantee that if we peek and then dequeue, we'll get the item we peeked?

If we have not full platform support, can we use `#if` preprocessing directives to use the best implementation possible for the more modern ones, and only take that risk on the older ones?